python-sdk/nuscenes/eval/detection/tests/test_main.py

```diff
@@ -116,7 +128,8 @@ def test_delta(self):
 
         # Score of 0.22082865720221012 was measured on the branch "release_v0.2" on March 7 2019.
         # After changing to measure center distance from the ego-vehicle this changed to 0.2199307290627096
-        self.assertAlmostEqual(metrics['weighted_sum'], 0.2199307290627096)
+        # After replacing attribute scores with single attribute, changed to 0.2147663200815183.
+        self.assertAlmostEqual(metrics['weighted_sum'], 0.2147663200815183)
```

Comment:
@holger-nutonomy : what is going on here? The change of format should not change the results. Was the previous implementation not to specification? : _"mean Average Attribute Error (mAAE): For each match we compute the attribute error as as 1 - acc, where acc is the attribute classification accuracy of all the relevant attributes of the ground-truth class. The attribute error is ignored for classes without attributes."_

Reply:
Or due to different generation of the `mock_results`?

Reply:
Yes, first we draw a random class, then we draw a random attribute. Before we generated random scores and took the relevant attribute with the highest score.

Reply:
Did you try back-porting the `mock_results` generation so that it's identical? Did that confirm that you got the same results?

Reply:
@holger-nutonomy : I added a back-port in https://github.com/nutonomy/nuscenes-devkit/pull/60/commits/9dc2f16cfe92b82c816502833a766b2d8a0495ea. This generates the same attributes as before but in the new format. With these changes we are very close to the results before this PR, but not quite there. Can you look into why?
- before_this_pr: `0.2199307290627096`
- with_your_new_test_gen_code: `0.2147663200815183`
- with_back-ported_test_gen_code: `0.22015241372993072`

Reply:
@holger-nutonomy : False alarm. I fixed the back-port in https://github.com/nutonomy/nuscenes-devkit/pull/60/commits/7c05deee6d87587a9bcffc8eb9a13745790e9787, and now everything checks out. 

I suggest we use this (uglier) fixture gen. code for now. We can make final clean-up later once we have done all changes.